test/persistence.async.test.js

```diff
@@ -368,13 +368,29 @@ describe('Persistence async', function () {
   })
 
   it('Can listen to compaction events', async () => {
+    d.insertAsync({ a: 2 })
     const compacted = new Promise(resolve => {
       d.once('compaction.done', function () {
         resolve()
       })
     })
     await d.compactDatafileAsync()
     await compacted // should already be resolved when the function returns, but still awaiting for it
+
+    const failure = new Promise(resolve => {
+      d.once('compaction.failed', function (error) {
+        resolve(error)
+      })
+    })
+
+    const afterSerializationOriginal = d.persistence.afterSerialization
+    d.persistence.afterSerialization = () => { throw new Error('synthetic error') }
+    await d.compactDatafileAsync().then(() => { throw new Error('should have failed') }, error => {
+      if (!error || !(error instanceof Error) || error.message !== 'synthetic error') throw error
+    })
+    const error = await failure
+    if (!error || !(error instanceof Error) || error.message !== 'synthetic error') throw error
+    d.persistence.afterSerialization = afterSerializationOriginal
```

Comment:
To make sure that this cleanup is always performed, it would be better to have this line in a `finallly` block.

Reply:
Test are independant, there is a beforeEach that resets everything, the final line is actually useless, I suggest to remove it rather than put it in a finally block.